app/models/container_group.rb

```diff
@@ -37,6 +38,25 @@ def ready_condition_status
     ready_condition.try(:status) || 'None'
   end
 
+  def containers_states_summary
+    containers = self.containers.to_a
+    {
+      :waiting    => containers_state_count('waiting', containers),
+      :running    => containers_state_count('running', containers),
+      :terminated => containers_state_count('terminated', containers)
```

Comment:
@zakiva iteration should happen once and have a hash of `{:waiting => count, :running => count, :terminated => count}` that then you can use to create what you need.